src/easylink/pipeline_schema.py

```diff
@@ -1,77 +1,47 @@
 from pathlib import Path
-from typing import Callable, List, Optional
+from typing import Any, List, Optional
 
-from easylink.step import Step
-from easylink.utilities.validation_utils import validate_input_file_dummy
+import networkx as nx
 
+from easylink.pipeline_schema_constants import ALLOWED_SCHEMA_PARAMS
+from easylink.step import CompositeStep, Step
 
-class PipelineSchema:
-    """Defines the allowable schema(s) for the pipeline."""
 
-    def __init__(self, name: str, validate_input: Callable) -> None:
-        self.name = name
-        self.validate_input = validate_input
-        self.steps = []
+class PipelineSchema(CompositeStep):
```

Comment:
The pipeline schema is one big "CompositeStep", as the top level of the composite step data structure.